packages/sharing-smart-contract/contracts/modules/Subscription.sol

```diff
@@ -41,6 +47,31 @@ contract Subscription is Collection {
     /***************************************************************************
      *                        Functions                                        *
      ***************************************************************************/
+    // can subscribe for an illimited duration ?
+    function subscribeTo(uint256 _collectionId) public payable returns (uint256) {
+        require(subscriptionParams[_collectionId].duration > 0, "Subscription parameters not set");
+        require(msg.value >= subscriptionParams[_collectionId].price, "Fund sent insufficient");
+
+        uint256 extraFunds = msg.value - subscriptionParams[_collectionId].price;
+        if (extraFunds > 0) {
+            (bool success, ) = msg.sender.call{value: extraFunds}("");
+            require(success, "Failed to send back extra funds");
+        }
+        uint48 endDate = uint48(block.timestamp) + subscriptionParams[_collectionId].duration;
+        subscribers[_collectionId][msg.sender] = endDate;
+        emit NewSubscription(msg.sender, endDate);
+        return endDate;
+    }
+
+    // est ce que cette fonctions met toute la collection a la souscription ou 1 protectedData par une ????
```

Comment:
set one protected data available in the subscription

Reply:
Commit: df316f0b1d73457229f2268817a9ba7ff5ce4c2b